modules/tinymce/src/core/main/ts/api/util/Tools.ts

```diff
@@ -8,22 +8,23 @@
 import * as ArrUtils from '../../util/ArrUtils';
 import Env from '../Env';
 
-type ArrayCallback<T, R> = (x: T, i: number, xs: ReadonlyArray<T>) => R;
-type ObjCallback<T, R> = (value: T[keyof T], key: string, obj: T) => R;
+type ArrayCallback<T, R> = ArrUtils.ArrayCallback<T, R>;
```

Comment:
```suggestion
type ArrayCallback = ArrUtils.ArrayCallback;
```
Shouldn't this work?

Reply:
> TS2314: Generic type 'ObjCallback' requires 2 type argument(s).

Reply:
I should really learn typescript some day huh

Reply:
I really do wish if TS allowed this though, would make things much easier.